From d7eb9186c64094d6413bf98220e04b5350d2292d Mon Sep 17 00:00:00 2001
From: Kevin Wolf <kwolf@redhat.com>
Date: Wed, 6 Nov 2013 14:41:14 +0100
Subject: [PATCH 38/81] cpu: Document why cannot_instantiate_with_device_add_yet

RH-Author: Kevin Wolf <kwolf@redhat.com>
Message-id: <1383748882-22831-10-git-send-email-kwolf@redhat.com>
Patchwork-id: 55535
O-Subject: [RHEL-7.0 qemu-kvm PATCH v2 09/17] cpu: Document why cannot_instantiate_with_device_add_yet
Bugzilla: 1001216
RH-Acked-by: Marcel Apfelbaum <marcel.a@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Kevin Wolf <kwolf@redhat.com>
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>

From: Markus Armbruster <armbru@redhat.com>

Signed-off-by: Markus Armbruster <armbru@redhat.com>
Reviewed-by: Peter Maydell <peter.maydell@linaro.org>
(cherry picked from pending upstream submission)
Signed-off-by: Kevin Wolf <kwolf@redhat.com>
---
 qom/cpu.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 qom/cpu.c |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/qom/cpu.c b/qom/cpu.c
index f97e1b8..859dae5 100644
--- a/qom/cpu.c
+++ b/qom/cpu.c
@@ -210,7 +210,11 @@ static void cpu_class_init(ObjectClass *klass, void *data)
     k->write_elf64_qemunote = cpu_common_write_elf64_qemunote;
     k->write_elf64_note = cpu_common_write_elf64_note;
     dc->realize = cpu_common_realizefn;
-    dc->cannot_instantiate_with_device_add_yet = true; /* FIXME explain why */
+    /*
+     * Reason: CPUs still need special care by board code: wiring up
+     * IRQs, adding reset handlers, halting non-first CPUS, ...
+     */
+    dc->cannot_instantiate_with_device_add_yet = true;
 }
 
 static const TypeInfo cpu_type_info = {
-- 
1.7.1

